tempodb/encoding/common/bloom.go

```diff
@@ -3,40 +3,53 @@ package common
 import (
 	"bytes"
 
-	"github.com/grafana/tempo/pkg/util"
 	"github.com/willf/bloom"
+
+	"github.com/grafana/tempo/pkg/util"
 )
 
-const shardNum = 10
+const legacyShardCount = 10
 
 type ShardedBloomFilter struct {
 	blooms []*bloom.BloomFilter
 }
 
-func NewWithEstimates(n uint, fp float64) *ShardedBloomFilter {
-	b := &ShardedBloomFilter{
-		blooms: make([]*bloom.BloomFilter, shardNum),
+// NewBloom creates a ShardedBloomFilter
+func NewBloom(fp float64, shardSize, estimatedObjects uint) *ShardedBloomFilter {
```

Comment:
how reliable is estimatedObjects? I feel like we do a pretty good job of passing a good value in here, but not 100% sure.

Reply:
This is accurate at the ingesters, but at the compactors the value is a sum of the object count present in the individual blocks.. so estimatedObjects will be > actualObjects

Reply:
i suppose that means we're going to be creating smaller bloom filters than expected and consequently more than we need.

easiest thing to do here may just be to push the change and see the impact in practice.

Reply:
The overage during typical compaction is probably a few percent, but when compacting level 0 blocks it will be much larger.  I.e. worst case is RF3 compacting 3 identical blocks, then it will be 3x needed.   Playing with a bloom filter calculator, keep constant filter size, but with 1/3rd the number of entries changes FP from 0.01 -> 0.004.

Reply:
Right. That's true. I think this is acceptable as well, level 0 blocks will be quickly crunched together, we can afford to have a lower fp and higher shards, should not affect performance.